src/synopsys-action/factory/github-client-service-factory.ts

```diff
@@ -0,0 +1,60 @@
+import {GithubClientServiceInterface} from '../service/github-client-service-interface'
+import {GithubClientServiceCloud} from '../service/impl/cloud/github-client-service-cloud'
+import {info} from '@actions/core'
+import * as constants from '../../application-constants'
+import {GithubClientServiceV1} from '../service/impl/enterprise/v1/github-client-service-v1'
+import {GithubClientServiceV2} from '../service/impl/enterprise/v2/github-client-service-v2'
+import {HttpClient} from 'typed-rest-client/HttpClient'
+import * as inputs from '../inputs'
+
+export const GitHubClientServiceFactory = {
+  SUPPORTED_VERSIONS_V1: ['3.11', '3.12'],
+  SUPPORTED_VERSIONS_V2: ['3.13', '3.14'],
+  DEFAULT_VERSION: '3.12',
+
+  async fetchVersion(githubServerUrl: string): Promise<string> {
+    const githubEnterpriseMetaUrl = '/meta'
+    const githubToken = inputs.GITHUB_TOKEN
+    const endpoint = githubServerUrl.concat(githubEnterpriseMetaUrl)
+
+    try {
+      const httpClient = new HttpClient('GitHubClientServiceFactory')
+      const httpResponse = await httpClient.get(endpoint, {
+        Authorization: `Bearer ${githubToken}`,
+        Accept: 'application/vnd.github+json'
+      })
+
+      if (httpResponse.message.statusCode === constants.HTTP_STATUS_OK) {
+        const metaDataResponse = JSON.parse(await httpResponse.readBody())
+        return metaDataResponse.installed_version
+      } else {
+        throw new Error(`No version info found for endpoint : ${endpoint}`)
+      }
+    } catch (error) {
+      throw new Error(`Fetching version info for enterprise server failed : ${error}`)
```

Comment:
we don't need to throw error if there in any issue in this API, we should use default version.
It may fail in the actual API call but we shouldn't fail here.

Reply:
updated